packages/flutter_tools/test/commands.shard/hermetic/doctor_test.dart

```diff
@@ -187,6 +189,66 @@ void main() {
       expect(message.message, startsWith('Flutter extension not installed'));
       expect(message.isError, isTrue);
     }, overrides: noColorTerminalOverride);
+
+    group('device validator', () {
+      testWithoutContext('no devices', () async {
+        final MockDeviceManager mockDeviceManager = MockDeviceManager();
+
+        when(mockDeviceManager.getAllConnectedDevices()).thenAnswer(
+          (Invocation invocation) => Future<List<Device>>.value(<Device>[])
+        );
+        when(mockDeviceManager.getDeviceDiagnostics()).thenAnswer(
+          (Invocation invocation) => Future<List<String>>.value(<String>[])
+        );
+
+        final DeviceValidator deviceValidator = DeviceValidator(mockDeviceManager, UserMessages());
+        final ValidationResult result = await deviceValidator.validate();
+        expect(result.type, ValidationType.notAvailable);
+        expect(result.messages.single.message, 'No devices available');
+        expect(result.messages.single.type, ValidationMessageType.hint);
+        expect(result.statusInfo, isNull);
+      });
+
+      testWithoutContext('diagnostic message', () async {
+        final MockDeviceManager mockDeviceManager = MockDeviceManager();
+
+        when(mockDeviceManager.getAllConnectedDevices()).thenAnswer(
+          (Invocation invocation) => Future<List<Device>>.value(<Device>[])
+        );
+        when(mockDeviceManager.getDeviceDiagnostics()).thenAnswer(
+          (Invocation invocation) => Future<List<String>>.value(<String>['Device locked'])
+        );
+
+        final DeviceValidator deviceValidator = DeviceValidator(mockDeviceManager, UserMessages());
+        final ValidationResult result = await deviceValidator.validate();
+        expect(result.type, ValidationType.notAvailable);
+        expect(result.messages.single.message, 'Device locked');
+        expect(result.messages.single.type, ValidationMessageType.hint);
+        expect(result.statusInfo, isNull);
+      });
+
+      testWithoutContext('diagnostic message and devices', () async {
+        final MockDeviceManager mockDeviceManager = MockDeviceManager();
+        final MockDevice mockDevice = MockDevice();
+
+        when(mockDeviceManager.getAllConnectedDevices()).thenAnswer(
+          (_) => Future<List<Device>>.value(<Device>[mockDevice])
+        );
+        when(mockDeviceManager.getDeviceDiagnostics()).thenAnswer(
+          (_) => Future<List<String>>.value(<String>['Device locked'])
+        );
+
+        final DeviceValidator deviceValidator = DeviceValidator(mockDeviceManager, UserMessages());
+        final ValidationResult result = await deviceValidator.validate();
+        expect(result.type, ValidationType.installed);
+        expect(result.messages.length, 2);
+        expect(result.messages.first.message, contains (mockDevice.id));
+        expect(result.messages.first.type, ValidationMessageType.information);
+        expect(result.messages[1].message, 'Device locked');
```

Comment:
I believe the message types have equality defined, so you could do:

```dart
expect(result.messages, [
  ValidationMessage(...),
  ValidationMessage(...),
]);
```

This will probably give a better error if the length changes